docs/training/apis-howto/api-core-mnist-ug.rst

```diff
@@ -0,0 +1,573 @@
+
+.. code::
+
+   #NEW: Get hparams chosen for this trial from cluster info object.
+   hparams = info.trial.hparams
+
+.. image:: /assets/images/model-def-adaptive-script.png
+
+Then, pass the hyperparameters into your model and optimizer:
+
+.. code::
+
+   #NEW: Pass relevant hparams to model and optimizer.
+   model = Net(hparams).to(device) optimizer =
+   optim.Adadelta(model.parameters(), lr=hparams[“learning_rate”])
+
+Ensure your model is set to use the selected values on a per-trial basis rather than your previously
+hardcoded values:
+
+..
+   note:
+
+   Refer to line 18 in ``model_def_adaptive.py``
+
+.. code::
+
+   class Net(nn.Module):
+       # NEW: Add hparams to **init**.
+       def **init**\ (self, hparams):
+           # NEW: Read hyperparameters provided for this trial.
+           super(Net, self).__init()
+           self.conv1 = nn.Conv2d(1,hparams[“n_filters1”], 3, 1)
+           self.conv2 = nn.Conv2d(hparams[“n_filters1”], hparams[“n_filters2”], 3, 1)
+           self.dropout1 = nn.Dropout(hparams[“dropout1”])
+           self.dropout2 = nn.Dropout(hparams[“dropout2”])
+           self.fc1 = nn.Linear(144 * hparams[“n_filters2”], 128)
+           self.fc2 = nn.Linear(128, 10)
+
+Step 4.1: Run the Experiment
+============================
+
+Run the following command to run the experiment:
+
+.. code::
+
+   det e create adaptive.yaml .
+
+In the Determined WebUI, navigate to the **Hyperparameters** tab.
+
+You should see a graph in the WebUI that displays the various trials initiated by the Adaptive ASHA
+hyperparameter search algorithm.
+
+******************************
+ Step 5: Distributed Training
+******************************
+
+The Core API has special considerations for running distributed training. Some of the more important
+considerations are:
+
+-  Access to all IP addresses of every node in the Trial (through the ClusterInfo API).
+
+-  Communication primitives such as :meth:`~determined.core.DistributedContext.allgather`,
+   :meth:`~determined.core.DistributedContext.gather`, and
+   :meth:`~determined.core.DistributedContext.broadcast` to give you out-of-the-box coordination
+   between workers.
+
+-  Since many distributed training frameworks expect all workers in training to operate in-step, the
+   :meth:`~determined.core.PreemptContext.should_preempt` call is automatically synchronized across
+   workers so that all workers decide to preempt or continue as a unit.
+
+..
+   tip Launchers
+
+   Typically, you do not have to write your own launcher.
+   Determined provides launchers for Horovod, torch.distributed, and
+   DeepSpeed. For more information about launcher options, visit
+   :doc:`Submit Experiment </training/submit-experiment>`.
+
+To perform distributed training with the Core API, you’ll need to use the appropriate distributed
+training library *before* creating a Determined :doc:`Distributed Context
+</reference/reference-training/training/api-core-reference>`.
```

Comment:
cross-links to api references should never render with the name a class but point to a broad document.  Just use autodoc references:
```
:class:`~determined.core.DistributedContext`
```
and let sphinx figure it out for you.

Reply:
resolved